Sources/ComposableArchitecture/Store.swift

```diff
@@ -212,17 +229,30 @@ public final class Store<State, Action> {
             self.send(fromLocalAction(localAction))
             localState = extractLocalState(self.state) ?? localState
             return .none
-          })
-
-        self.$state.producer
-          .startWithValues { [weak localStore] state in
+          }
+        )
+        
+        let observer = Signal<State, Never>.Observer(
+          value: { [weak localStore] state in
             guard let localStore = localStore else { return }
             localStore.state = extractLocalState(state) ?? localStore.state
+          },
+          failed: .none,
+          completed: { [weak localStore] in
+            localStore?.parentDisposable?.dispose()
+            localStore?.parentDisposable = nil
+          },
+          interrupted: { [weak localStore] in
+            localStore?.parentDisposable?.dispose()
+            localStore?.parentDisposable = nil
           }
+        )
+        
+        localStore.parentDisposable = self.$state.producer.start(observer)
```

Comment:
Is there any advantage to using an explicit `Signal.Observer` and disposing on `completed` and `interrupted`, instead of just:

```swift
localStore.parentDisposable = self.$state.producer
   .startWithValues { ... }
```

Surely the `Disposable` is disposed in any case if the stream completes or is interrupted?

Reply:
I mentioned it here https://github.com/trading-point/reactiveswift-composable-architecture/pull/39#issuecomment-815496441

Reply:
Oh sorry, I didn't see that 🙈  See here: https://github.com/trading-point/reactiveswift-composable-architecture/pull/39#discussion_r609938905

I tested it with just `startWithValues` and it's fine.

Reply:
@maximkrouk this one can also be change to just use `startWithValues` instead of an explicit `Observer`.

Thanks for changing the others 👍

Reply:
Oh, missed it, gonna fix it today too

Reply:
Done ✅